File: HaskellCocoa/Cocoa/AppKit/HNSScrollView.hs



{-# LANGUAGE ForeignFunctionInterface #-}



module Cocoa.AppKit.HNSScrollView
(
    HNSScrollView
,   HNSScrollViewObj
,   nsScrollViewCreate
,   nsSCrollView_contentView
,   nsScrollView_setDocumentView
,   nsScrollView_documentView
,   nsScrollView_setHasHorizontalScroller
,   nsScrollView_setHasVerticalScroller
)
where


import Foreign
import Cocoa.Runtime.HNSObject
import Cocoa.AppKit.HNSView



foreign import ccall hns_ScrollViewCreate :: IO (Ptr a)
foreign import ccall hns_scrollView_contentView :: Ptr a -> IO (Ptr b)
foreign import ccall hns_scrollView_setDocumentView :: Ptr a -> Ptr b -> IO ()
foreign import ccall hns_scrollView_documentView :: Ptr a -> IO (Ptr b)

foreign import ccall hns_scrollView_setHasHorizontalScroller :: Ptr a -> Bool -> IO ()
foreign import ccall hns_scrollView_setHasVerticalScroller :: Ptr a -> Bool -> IO ()




class (HNSView a) => HNSScrollView a where

    nsSCrollView_contentView :: HNSView b => Ptr a -> IO (Ptr b)
    nsSCrollView_contentView = hns_scrollView_contentView

    nsScrollView_setDocumentView :: HNSView b =>  Ptr a -> Ptr b -> IO ()
    nsScrollView_setDocumentView = hns_scrollView_setDocumentView

    nsScrollView_documentView :: HNSView b => Ptr a -> IO (Ptr b)
    nsScrollView_documentView = hns_scrollView_documentView

    nsScrollView_setHasHorizontalScroller :: Ptr a -> Bool -> IO ()
    nsScrollView_setHasHorizontalScroller = hns_scrollView_setHasHorizontalScroller

    nsScrollView_setHasVerticalScroller :: Ptr a -> Bool -> IO ()
    nsScrollView_setHasVerticalScroller = hns_scrollView_setHasVerticalScroller





data HNSScrollViewObj

instance HNSObject HNSScrollViewObj where
instance HNSView HNSScrollViewObj where
instance HNSScrollView HNSScrollViewObj where



nsScrollViewCreate :: IO (Ptr HNSScrollViewObj)
nsScrollViewCreate = hns_ScrollViewCreate

File: HaskellCocoa/Cocoa/AppKit/HNSScrollView.m
//
//  HNSScrollView.m
//
//
//  Created by middleware on 2/26/16.
//
//

#import <AppKit/AppKit.h>

#import "HNSObject.h"
#import "HNSScrollView.h"


HsPtr hns_ScrollViewCreate(void)
{
    NSScrollView* view = [[NSScrollView alloc] init];
    
    hns_retain(view);
    return (__bridge HsPtr)(view);
}



HsPtr hns_scrollView_contentView(HsPtr view)
{
    NSScrollView* pView = (__bridge NSScrollView*)view;
    return (__bridge HsPtr)pView.contentView;
}


void hns_scrollView_setDocumentView(HsPtr view, HsPtr docView)
{
    NSScrollView* pView = (__bridge NSScrollView*)view;
    NSView* pDocView = (__bridge NSView*)docView;
    [pView setDocumentView:pDocView];
}



void hns_scrollView_setHasHorizontalScroller(HsPtr view, bool b)
{
    NSScrollView* pView = (__bridge NSScrollView*)view;
    [pView setHasHorizontalScroller:b];
}


void hns_scrollView_setHasVerticalScroller(HsPtr view, bool b)
{
    NSScrollView* pView = (__bridge NSScrollView*)view;
    [pView setHasVerticalScroller:b];
}


